File: 3355_Zero_Array_Transformation_I.py
class Solution(object):
    def isZeroArray(self, a, b):
        d = [0]*(len(a)+2)
        for l, r in b:
            d[l] += 1
            d[r+1] -= 1
        for i in range(1, len(a)): d[i] += d[i-1]
        return all(d[i] >= a[i] for i in range(len(a)))

File: 3355_Zero_Array_Transformation_I.cpp
class Solution {
public:
    bool isZeroArray(vector<int>& q, vector<vector<int>>& r) {
        int n = q.size();
        map<int, int> m;
        for (auto& v : r) ++m[v[0]], --m[v[1]+1];
        int c = 0;
        for (int i = 0; i < n; ++i) {
            c += m[i];
            if (c < q[i]) return 0;
        }
        return 1;
    }
};
